hydrabooster/node/node.go

```diff
@@ -0,0 +1,108 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"time"
+
+	ipns "github.com/ipfs/go-ipns"
+	libp2p "github.com/libp2p/go-libp2p"
+	circuit "github.com/libp2p/go-libp2p-circuit"
+	connmgr "github.com/libp2p/go-libp2p-connmgr"
+	crypto "github.com/libp2p/go-libp2p-core/crypto"
+	host "github.com/libp2p/go-libp2p-core/host"
+	peer "github.com/libp2p/go-libp2p-core/peer"
+	dht "github.com/libp2p/go-libp2p-kad-dht"
+	dhtopts "github.com/libp2p/go-libp2p-kad-dht/opts"
+	record "github.com/libp2p/go-libp2p-record"
+	hyopts "github.com/libp2p/hydra-booster/hydrabooster/opts"
+	"github.com/multiformats/go-multiaddr"
+)
+
+func randBootstrapAddr(bootstrapPeers []multiaddr.Multiaddr) (*peer.AddrInfo, error) {
+	addr := bootstrapPeers[rand.Intn(len(bootstrapPeers))]
+	return peer.AddrInfoFromP2pAddr(addr)
+}
+
+// BootstrapStatus describes the status of connecting to a bootstrap node
+type BootstrapStatus struct {
+	Done bool
+	Err  error
+}
+
+// HydraNode is a container for libp2p components used by a Hydra Booster node
+type HydraNode struct {
+	Host host.Host
+	DHT  *dht.IpfsDHT
+}
+
+// NewHydraNode constructs a new Hydra Booster node
+func NewHydraNode(options ...hyopts.Option) (*HydraNode, chan BootstrapStatus, error) {
+	cfg := hyopts.Options{}
+	cfg.Apply(append([]hyopts.Option{hyopts.Defaults}, options...)...)
+
+	cmgr := connmgr.NewConnManager(1500, 2000, time.Minute)
```

Comment:
Make the connManager min and max be const at the top (so that they are easy to change)?